### DISCUSSION TITLE
are there any plans to replace Vite with rsbuild?

### DISCUSSION THREAD

**Discussion by igotfr** [solidjs/solid-start#1694] [Q&A] [answered]
just a curiosity

**Comment by lxsmnsyc** [comment]
I don't think this will ever be consideration.

**Comment by atk** [comment, marked as answer]
Solid is based on a babel transform that turns the JSX into templates and reactive expressions (dom-expressions used inside babel-preset-solid). It is included by vite-plugin-solid to be used by vite to run and bundle solid code.

To use rsbuild, we would need to develop and maintain another plugin specifically for that. Unless esbuild gets a wider adoption or has an advantage that would make it more desirable than vite to the maintainers, any official solid-esbuild plugin is highly improbable.

**Reply by igotfr** [reply]
the advantage of rsbuild is your supposed speed (I didn't test), but there are certainly more things to consider

**Reply by rhengles** [reply]
Vite is already pretty darn fast, and it is accelerating even further in the future.

**Reply by igotfr** [reply]
apparently there is the possibility of being faster
